R/types.R

```diff
@@ -685,7 +685,7 @@ is_copyable <- function(x) {
 is_equal <- function(x, y) {
   identical(x, y)
 }
-#' @useDynLib rlang rlang_is_identical
-is_identical <- function(x, y) {
-  .Call(rlang_is_identical, x, y)
+#' @useDynLib rlang rlang_is_reference
+is_reference <- function(x, y) {
```

Comment:
that's a good idea. I think one goal of rlang should be to make it easy to prototype C-side code. This should also help new developers get acquainted with R internals (e.g. playing with the `node_` functions to explore the structure of expressions).